main/src/ca/uwaterloo/flix/language/phase/Reader.scala

```diff
@@ -59,7 +63,9 @@ object Reader {
               result += (src -> ())
             }
 
-          case Input.Unknown => throw InternalCompilerException("Impossible to read source code from unknown input.", SourceLocation.Unknown)
+          case Input.FileInPackage(_, _, _) => throw InternalCompilerException("Impossible.", SourceLocation.Unknown)
+
+          case Input.Unknown => throw InternalCompilerException("Impossible.", SourceLocation.Unknown)
```

Comment:
Not so helpful error messages :)

Reply:
I think its because our modelling is wrong and these things should go away.